README.md

```diff
@@ -1,2 +1,209 @@
+    orchset_scores[track_id] = score
+
+# Split the results by composer and by instrumentation
+composer_scores = {}
+strings_no_strings_scores = {True: {}, False: {}}
+for track_id, track_data in orchset_data.items():
+    if track_data.composer not in composer_scores.keys():
+        composer_scores[track_data.composer] = {}
+
+    composer_scores[track_data.composer][track_id] = orchset_scores[track_id]
+    strings_no_strings_scores[track_data.contains_strings][track_id] = \
+        orchset_scores[track_id]
+
+```
+
+
+## Contributing a new dataset loader
+
+To add a new dataset loader:
+1. Create an index in `mir_dataset_loaders/indexes`, e.g. `unicorn_index.json`
+2. Create a module in `mir_dataset_loaders`, .e.g `unicorn.py`
+
+
+### Creating an index
+
+The index should be a json file where the top level keys are the unique track
+ids of the datset. The values should be a dictionary of metadata, and the keys
+of these dictionaries should be identical across the dataset.
+
+Any file path included should be relative to the top level directory of the dataset.
+
+Any unavailable field should be indicated with `null`.
+
+For example:
+```json
+{
+    'track1': {
+        'audio_path': 'Unicorn/track1.wav',
+        'color': 'red',
+        'genre': 'Electronica',
+        'sign': 'positive'
+    },
+    'track2': {
+        'audio_path': 'Unicorn/track2.wav',
+        'color': 'green',
+        'genre': 'Free Jazz',
+        'sign': 'negative'
+    },
+    'track3': {
+        'audio_path': 'Unicorn/track3.wav',
+        'color': 'purple',
+        'genre': 'Rockabilly',
+        'sign': null
+    }
+}
+```
+
+Notebooks used to create some of the dataset indexes are in the [notebooks]() folder.
+
+
+### Creating a module.
+
+copy paste this template and fill it in for your dataset:
+
+```python
+
+from collections import namedtuple
+import json
+
+from . import UNICORN_INDEX_PATH
+from .load_utils import validator
+
+UNICORN_INDEX = json.load(open(UNICORN_INDEX_PATH, 'r'))
+
+UnicornTrack = namedtuple(
+    'UnicornTrack',
+    ['track_id',
+     'TODO']
+)
+
+
+def download(data_home=None, clobber=False):
+    # TODO, write this, see orchset.py for an example
+    pass
+
+
+def validate(data_home):
+    file_keys = [TODO]  # keys in UNICORN_INDEX that are filepaths
+    missing_files = validator(UNICORN_INDEX, file_keys, data_home)
+    return missing_files
+
+
+def track_ids():
+    return list(UNICORN_INDEX.keys())
+
+
+def load(data_home=None):
+    validate(data_home)
+    data_dict = {}
+    for key in track_ids():
+        data_dict[key] = load_track(key, data_home=data_home)
+    return data_dict
+
+
+def load_track(track_id, data_home=None):
+    if track_id not in UNICORN_INDEX.keys():
+        raise ValueError(
+            "{} is not a valid track ID in Unicorn".format(track_id))
+    track_data = UNICORN_INDEX[track_id]
+
+    return UnicornTrack(
+        track_id,
+        TODO
+    )
```

Comment:
Will have to double check the most consistent way to do this as per `sklearn` and other libraries in research, this technically lends itself very well to a base class and extending e.g. `class UnicornModule(MirModule)` where `MirModule` is something you've provided to the users and `UnicornModule` is something they write themselves. If you do it this way, the consistent logic around interpreting your json is very easy to consolidate.

Reply:
Curious if @andreasjansson has thoughts